### DISCUSSION TITLE
셀렉터를 상수화해서 쓰는 것이 반드시 유용할까요?

### DISCUSSION THREAD

**Discussion by wonsss** [woowacourse-study/2022-code-review-study#3] [Lotto] [answered]
어느 리뷰어님께서 "많은 크루분들이 이렇게 셀렉터를 상수화해서 쓰고 계시던데, 개인적으로 이렇게 할 가치가 있는지 의문이에요."라는 말씀을 해주셨는데요. 같이 한 번 고민해보면 좋을 것 같아요. 장단점으로 무엇이 있을까요? 또는 어떤 상황에 적합 또는 부적합할까요?

**Comment by lokba** [comment]
제 생각을 한번 애기해보겠습니다.

저는 제 나름대로의 규칙을 만들었습니다.
selector에 접근하는 곳이 여러 곳인가? 아니면 한 곳인가? 입니다.

전자의 상황이라면, 저는 상수화를 진행할 것 같습니다. 
전자의 경우, 갑작스럽게 selector의 변경이 생긴다고 가정을 합시다. 상수화를 적용하면, 유지보수하기가 편합니다.
상수화를 적용하지 않는다면, 직접 일일히 변경을 해야하는 의미없는 시간을 소모하게 됩니다.

후자의 상황이라면, 저는 상수화를 진행하지 않습니다.
여러 곳에서 접근도 하지 않고, selector는 민감한 정보도 아니기에 굳이, 상수화를 적용할 필요가 있나 싶습니다.

**Reply by wonsss** [reply]
"selector에 접근하는 곳이 여러 곳인가? 아니면 한 곳인가?"
좋은 기준이네요!

**Comment by moonheekim0118** [comment, marked as answer]
저는 개인적으로 만드는 프로그램의 성격에 따라서 필요성이 달라질 것 같은데요,
만약 일회성으로 사용되거나, 유지보수가 필요없고 E2E 테스트코드도 필요없는 프로그램이라면 셀렉터를 굳이 시간들여서 상수화 할 필요는 없다고 생각해요. 오히려 개발의 효율성을 낮출 것 같네요.
근데 프로그램이 크고, 유지보수가 계속 될 예정이며, E2E테스트코드도 작성되어야 한다면 상수화를 하는게 멀게 봤을 때 이득이라고 생각하거든요. 나중에 개발자가 바뀌어도 상수화된 셀렉터만 보고 코드를 고칠 수 있으니까요.
그래서 개인적으로는 미션에서는 셀렉터를 상수화하는 습관을 들이는게 나쁘지는 않다고 봅니다.

**Comment by byhhh2** [comment]
저는 상수화에 반대하는 입장인데요. 
사실상 하나의 dom element를 여러 view에서 접근할 일이 없을 것 같아요. 
여러 view에서 하나의 dom element를 접근한다면 그건 view가 제대로 나눠지지 않았다는 뜻 같아요. 🤔
